src/streamlink/plugins/ceskatelevize.py

```diff
@@ -5,131 +5,178 @@
-        self.title = data["channelName"]
-
-        _hash = self.session.http.get(
-            "https://www.ceskatelevize.cz/v-api/iframe-hash/",
-            schema=validate.Schema(str))
-        res = self.session.http.get(
-            "https://www.ceskatelevize.cz/ivysilani/embed/iFramePlayer.php",
-            params={
-                "hash": _hash,
-                "origin": "iVysilani",
-                "autoStart": "true",
-                "videoID": video_id,
-            },
-        )
-
-        m = self._re_playlist_info.search(res.text)
-        if not m:
-            return
-        _type, _id = m.groups()
-
-        data = self.session.http.post(
+    def get_stream_url(self, video_id):
+        url = self.session.http.post(
             "https://www.ceskatelevize.cz/ivysilani/ajax/get-client-playlist/",
             data={
-                "playlist[0][type]": _type,
-                "playlist[0][id]": _id,
+                "playlist[0][type]": "channel",
+                "playlist[0][id]": video_id,
                 "requestUrl": "/ivysilani/embed/iFramePlayer.php",
                 "requestSource": "iVysilani",
                 "type": "html",
                 "canPlayDRM": "false",
             },
-            headers={
-                "x-addr": "127.0.0.1",
-            },
             schema=validate.Schema(
                 validate.parse_json(),
                 {
-                    validate.optional("streamingProtocol"): str,
-                    "url": validate.any(
-                        validate.url(),
-                        "Error",
-                        "error_region"
-                    )
-                }
+                    "url": validate.url(),
+                },
+                validate.get("url"),
             ),
         )
 
-        if data["url"] in ["Error", "error_region"]:
-            log.error("This stream is not available")
-            return
+        return self.session.http.get(
+            url,
+            schema=validate.Schema(
+                validate.parse_json(),
+                self.schema_playlist,
+                validate.get(("playlist", 0, "streamUrls", "main")),
+            ),
+        )
+
+    def get_ct24(self):
+        self.id = 24
+        self.title = "ČT24"
+        return self.get_stream_url(self.id)
+
+    def get_decko(self):
+        self.id = 5
+        self.title = "Déčko"
+        return self.get_stream_url(self.id)
+
+    def get_sport(self, content):
+        video_id, key, date = validate.Schema(
+            validate.parse_html(),
+            validate.xml_xpath_string(".//section[@id='live']/@data-ctcomp-data"),
+            str,
+            validate.parse_json(),
+            {
+                "items": [{
+                    "items": [{
+                        validate.optional("video"): {
+                            "data": {
+                                "source": {
+                                    "playlist": [{
```

Comment:
`404 Client Error: Not found for url`

use **noDrmData**

```python
    def get_sport(self, content):
        video_id, key, date = validate.Schema(
            validate.parse_html(),
            validate.xml_xpath_string(".//section[@id='live']/@data-ctcomp-data"),
            str,
            validate.parse_json(),
            {
                "items": [{
                    "items": [{
                        validate.optional("video"): {
                            "data": {
                                "source": {
                                    "playlist": [{
                                        "id": int,
                                        "key": str,
                                        "date": str,
                                        "noDrmData": {
                                            "id": int,
                                            "key": str,
                                            "drm": int,
                                            "quality": str,
                                            "assetId": str,
                                        }
                                    }],
                                },
                            },
                        },
                    }],
                }],
            },
            validate.get(("items", 0, "items", 0, "video", "data", "source", "playlist", 0)),
            validate.union_get(("noDrmData", "id"), ("noDrmData", "key"), "date"),
        ).validate(content)

        self.id = video_id
        self.title = "ČT sport"
```

Reply:
Thanks.